homeassistant/requirements.py

```diff
@@ -240,24 +251,41 @@ async def _async_process_integration(
             raise exceptions[0]
 
     async def async_process_requirements(
-        self, name: str, requirements: list[str]
+        self, name: str, requirements: list[str], is_built_in: bool
     ) -> None:
         """Install the requirements for a component or platform.
 
         This method is a coroutine. It will raise RequirementsNotFound
         if an requirement can't be satisfied.
         """
-        if self.hass.config.skip_pip_packages:
-            skipped_requirements = {
-                req
-                for req in requirements
-                if Requirement(req).name in self.hass.config.skip_pip_packages
+        if DEPRECATED_PACKAGES or self.hass.config.skip_pip_packages:
+            all_requirements = {
+                requirement_string: Requirement(requirement_string)
+                for requirement_string in requirements
             }
-
-            for req in skipped_requirements:
-                _LOGGER.warning("Skipping requirement %s. This may cause issues", req)
-
-            requirements = [r for r in requirements if r not in skipped_requirements]
+            if DEPRECATED_PACKAGES:
+                for requirement_string, requirement_details in all_requirements.items():
+                    if deprecation := DEPRECATED_PACKAGES.get(requirement_details.name):
+                        _LOGGER.warning(
+                            "Detected that %sintegration '%s' %s. %s %s",
+                            "" if is_built_in else "custom ",
+                            name,
+                            f"has requirement '{requirement_string}' which {deprecation[0]}",
+                            f"This will stop working in Home Assistant {breaks_in_ha_version}, please"
+                            if (breaks_in_ha_version := deprecation[1])
+                            else "Please",
+                            async_suggest_report_issue(
+                                self.hass, integration_domain=name
+                            ),
+                        )
+            if skip_pip_packages := self.hass.config.skip_pip_packages:
+                for requirement_string, requirement_details in all_requirements.items():
+                    if requirement_details.name in skip_pip_packages:
+                        _LOGGER.warning(
+                            "Skipping requirement %s. This may cause issues",
+                            requirement_string,
+                        )
+                        requirements.remove(requirement_string)
```

Comment:
It looks like we are mutating the passed list now. I think we just need to make a copy